### ISSUE TITLE
Tests fails on windows due to endline

### ISSUE THREAD

**Issue by mapgccv** [ampl/api-test#1]
Some tests fail on Windows due to problems in handling newline. E.g.: EntityTest.testGetInvalidEntity
```Java
  assertEquals(
        "1 indices for set of dimen 2\n" +
        "Line: 1, Offset: 28, Context:\n" +
 "var x{i in {(1, 2), ('abc', 3)}} >= i;\n", out.toString());
```
fails as in windows as the output uses \r\n as newline.

**Comment by vitaut** [comment]
Good catch. I think it should be handled on the API level in the same way it is done in the standard I/O API, e.g. when you open a file in text mode.

**Comment by mapgccv** [comment]
Two comments:

1 Isn't it better to have platform dependent newlines anyway? 
2 Independently: fair enough for some of those (where the API is artificially adding those newlines), but for some (where the ByteArrayOutputStream TestBase.out is used), I think the newline is actually converted by the ByteArrayOutputStream.toString() method.

**Comment by vitaut** [comment]
> 1 Isn't it better to have platform dependent newlines anyway? 

Newlines are platform-dependent on a file level. On a language level newlines are usually represented in a platform-independent fashion. For example, in idiomatic C and C++ you write:

```c
printf("test\n");
```
and not 

```c
printf("test\r\n");
```
The translation into system-dependent line ending is transparent to the user which makes the user code simpler.

> 2 Independently: fair enough for some of those (where the API is artificially adding those newlines), but for some (where the ByteArrayOutputStream TestBase.out is used), I think the newline is actually converted by the ByteArrayOutputStream.toString() method.

I agree, output is a file-level API and should probably support platform-specific newlines.

**Comment by vitaut** [comment]
Part (2) should be fixed in https://github.com/ampl/api-test/commit/1e634fca4813757c077d0076f135cb276cba0f06.

**Comment by mapgccv** [comment]
The tests (besides the CBC issue) run successfully on API 0.2.5.

**Comment by vitaut** [comment]
Cool.

It's unrelated, but could you do something about the sphinx submodule:

```
[submodule "cpp/thirdparty/sphinx-1.2.3"]
	path = cpp/thirdparty/sphinx-1.2.3
	url = git@callisto:sphinx.git
```

as it causes the following error:
```
Cloning into 'cpp/thirdparty/sphinx-1.2.3'...
ssh: Could not resolve hostname callisto: Name or service not known
```

BTW [Sphinx repo](https://github.com/sphinx-doc/sphinx) is available on GitHub.

**Comment by mapgccv** [comment]
Thanks, 
it appears i don't need it anymore. The problem arose because of breathe not supporting sphinx 1.3, so I kept a clone of  the latest working version there.. Now it appears that... one of the maintainers has implemented 1.3 support, so we can rid of that.
